akka-projection-slick/src/main/scala/akka/projection/slick/internal/SlickProjectionImpl.scala

```diff
@@ -83,163 +77,32 @@ private[projection] class SlickProjectionImpl[Offset, Envelope, P <: JdbcProfile
-
-      // this will count as one attempt
-      val firstAttempt = futureCallback()
-
-      firstAttempt.recoverWith {
-        case NonFatal(err) =>
-          val failedOffset = sourceProvider.extractOffset(envelope)
-
-          handler.onFailure(envelope, err) match {
-            case Fail =>
-              logger.error(
-                cause = err,
-                template =
-                  "Failed to process envelope with offset [{}]. Projection will stop as defined by recovery strategy.",
-                failedOffset)
-              firstAttempt
-
-            case Skip =>
-              logger.warning(
-                template =
-                  "Failed to process envelope with offset [{}]. Envelope will be skipped as defined by recovery strategy. Exception {}({})",
-                failedOffset,
-                err.getClass.getSimpleName,
-                err.getMessage)
-              futDone
-
-            case RetryAndFail(retries, delay) =>
-              val totalRetries = retries - 1
-              logger.error(
-                cause = err,
-                template = "Failed to process envelope with offset [{}] failed. Will retry {} time(s).",
-                failedOffset,
-                totalRetries)
-
-              // less one attempt
-              val retied = retry(futureCallback, totalRetries, delay)
-              retied.onComplete {
-                case Failure(exception) =>
-                  logger.error(
-                    cause = exception,
-                    template =
-                      "Failed to process envelope with offset [{}] after {} retries. Projection will stop as defined by recovery strategy.",
-                    failedOffset,
-                    totalRetries)
-                case _ => // do nothing
-              }
-              retied
-
-            case RetryAndSkip(retries, delay) =>
-              // less one attempt
-              val totalRetries = retries - 1
-              logger.warning(
-                template =
-                  "Failed to process envelope with offset [{}] failed. Will retry {} time(s). Exception {}({})",
-                failedOffset,
-                totalRetries,
-                err.getClass.getSimpleName,
-                err.getMessage)
-
-              retry(futureCallback, totalRetries, delay).recoverWith {
-                case NonFatal(exception) =>
-                  logger.warning(
-                    template =
-                      "Failed to process envelope with offset [{}] after {} retries. Envelope will be skipped as defined by recovery strategy. Last exception {}({})",
-                    failedOffset,
-                    totalRetries,
-                    exception.getClass.getSimpleName,
-                    exception.getMessage)
-                  futDone
-              }
-          }
-      }
-    }
-
-    def aroundUserHandler(env: Envelope)(handlerFunc: Envelope => slick.dbio.DBIO[Done]) = {
```

Comment:
I removed this. I think the purpose was to deal with thrown exceptions, but that is now handled in same way as if the handler returned Future.failed. There were no tests for this and I don't think it was completely correct anyway.